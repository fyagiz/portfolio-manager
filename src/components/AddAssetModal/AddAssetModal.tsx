import { Modal, Text, View } from "react-native";
import { AddAssetModalPropsType } from "./AddAssetModal.type";
import styles from "./AddAssetModal.style";

const AddAssetModal = (props: AddAssetModalPropsType) => {
  const { modalProps } = props;
  return (
    <Modal {...modalProps}>
      <View style={styles.container}>
        <View style={{ backgroundColor: "white", borderRadius: 4 }}>
          <Text>Example Modal</Text>
        </View>
      </View>
    </Modal>
  );
};

export default AddAssetModal;
