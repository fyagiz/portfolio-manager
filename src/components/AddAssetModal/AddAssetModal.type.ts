import { ModalProps } from "react-native";

export type AddAssetModalPropsType = {
  modalProps: ModalProps;
};
