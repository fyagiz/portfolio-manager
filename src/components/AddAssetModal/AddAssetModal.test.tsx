import { render } from "@testing-library/react-native";
import AddAssetModal from "./AddAssetModal";

/*
1- AddAssetModal should be rendered
*/

describe("AddAssetModal Unit Tests", () => {
  test("should render correctly", () => {
    const addAssetModal = render(<AddAssetModal modalProps={{ visible: true }} />);
    expect(addAssetModal).toMatchSnapshot();
  });
});
