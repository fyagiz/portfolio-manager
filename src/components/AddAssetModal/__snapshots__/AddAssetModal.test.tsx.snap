// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddAssetModal Unit Tests should render correctly 1`] = `
<Modal
  hardwareAccelerated={false}
  visible={true}
>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "white",
        }
      }
    >
      <Text>
        Example Modal
      </Text>
    </View>
  </View>
</Modal>
`;
